pkg/controller/stack/stack_controller.go

```diff
@@ -361,33 +392,59 @@ func (r *ReconcileStack) Reconcile(ctx context.Context, request reconcile.Reques
 		}
 	}
 
-	// If a branch is specified, then track changes to the branch.
-	trackBranch := len(sess.stack.Branch) > 0
+	// Proceed/Requeue logic: this depends on the kind of source, but broadly:
+	// - if the fetched revision is the same as the last one, proceed only if
+	//   `ContinueResyncOnCommitMatch`
+	// - if not proceeding, requeue in ResyncFrequencySeconds (sic)
 
+	// Does this need to be requeued to poll its source again?
```

Comment:
Might want a TODO here

Reply:
I'll rewrite the comment to be less ambiguous -- it's actually saying "This variable records whether the object needs to be requeued to poll its source"

Reply:
(I don't like this control flow in general, it was just the simplest change that we can fit both the new bits of code into conveniently)